ivy_tests/test_ivy/helpers.py

```diff
@@ -1303,69 +1321,94 @@ def test_method(
-        kwargs_np=constructor_kwargs_np,
+    args_constructor, kwargs_constructor, _, _, _ = create_args_kwargs(
+        args_np=args_np_constructor,
+        kwargs_np=kwargs_np_constructor,
         input_dtypes=input_dtypes_constructor,
         as_variable_flags=as_variable_flags_constructor,
     )
+    args_np_method, kwargs_np_method = kwargs_to_args_n_kwargs(
+        num_positional_args=num_positional_args_method, kwargs=all_as_kwargs_np_method
+    )
+    args_method, kwargs_method, _, _, _ = create_args_kwargs(
+        args_np=args_np_method,
+        kwargs_np=kwargs_np_method,
+        input_dtypes=input_dtypes_method,
+        as_variable_flags=as_variable_flags_method,
+    )
     # run
-    ins = ivy.__dict__[class_name](*constructor_args, **constructor_kwargs)
+    ins = ivy.__dict__[class_name](*args_constructor, **kwargs_constructor)
     ret, ret_np_flat = get_ret_and_flattened_np_array(
-        ins, *calling_args, **calling_kwargs
+        ins.__getattribute__(method_name), *args_method, **kwargs_method
     )
     # compute the return with a Ground Truth backend
     ivy.set_backend(ground_truth_backend)
-    calling_args_gt, calling_kwargs_gt, _, _, _ = create_args_kwargs(
-        args_np=calling_args_np,
-        kwargs_np=calling_kwargs_np,
-        input_dtypes=input_dtypes,
-        as_variable_flags=as_variable_flags,
-    )
-    constructor_args_gt, constructor_kwargs_gt, _, _, _ = create_args_kwargs(
-        args_np=constructor_args_np,
-        kwargs_np=constructor_kwargs_np,
+    args_gt_constructor, kwargs_gt_constructor, _, _, _ = create_args_kwargs(
+        args_np=args_np_constructor,
+        kwargs_np=kwargs_np_constructor,
         input_dtypes=input_dtypes_constructor,
         as_variable_flags=as_variable_flags_constructor,
     )
-    ins_gt = ivy.__dict__[class_name](*constructor_args_gt, **constructor_kwargs_gt)
+    args_gt_method, kwargs_gt_method, _, _, _ = create_args_kwargs(
+        args_np=args_np_method,
+        kwargs_np=kwargs_np_method,
+        input_dtypes=input_dtypes_method,
+        as_variable_flags=as_variable_flags_method,
+    )
+    ins_gt = ivy.__dict__[class_name](*args_gt_constructor, **kwargs_gt_constructor)
     ret_from_gt, ret_np_from_gt_flat = get_ret_and_flattened_np_array(
-        ins_gt, *calling_args_gt, **calling_kwargs_gt
+        ins_gt.__getattribute__(method_name), *args_gt_method, **kwargs_gt_method
     )
     ivy.unset_backend()
     # assuming value test will be handled manually in the test function
     if not test_values:
         return ret, ret_from_gt
     # value test
+    if test_values == "with_v":
+        test_values = (
+            True if "v" in kwargs_np_constructor or "v" in kwargs_np_method else False
+        )
     value_test(
         ret_np_flat=ret_np_flat,
         ret_np_from_gt_flat=ret_np_from_gt_flat,
-        rtol=rtol,
-        atol=atol,
+        rtol=rtol_,
+        atol=atol_,
+        test_values=test_values,
```

Comment:
as discussed above, we should simply add `value_test` inside an `if` statement here, rather than passing `test_values` as an argument.